Fix blake3 build with incompatible assemblers

thestr4ng3r opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Apple Clang 4.2 on Mac OS X 10.7 Lion does support avx2, but it has issues with some of the asm syntax used, in particular jc and some mov syntax.
So we add an additional check for a short assembly snippet.

**Test plan**

Build on x86_64 platforms should recognize simd extensions as before (can be checked in the CI output), but building on Lion should not fail.

XVilka commented on subprojects/packagefiles/blake3/unix_syntax_check.S:
```diff
@@ -0,0 +1,30 @@
+        mov r10d, dword ptr [rsp+0x110+8*rax]
+        jc 3f
+        nop
+		nop
```
Is broken alignment on purpose?

thestr4ng3r replied:
No, just me typing tabs, fixed.

thestr4ng3r committed:
Fix blake3 build with incompatible assemblers

Apple Clang 4.2 on Mac OS X 10.7 Lion does support avx2, but it has
issues with some of the asm syntax used, in particular jc and some mov
syntax.
So we add an additional check for a short assembly snippet.

XVilka reviewed (approved):
LGTM if CI will be green

wargio commented:
I throw it just in a simpler way. how about we just have an option? we try with auto, otherwise we provide other optional values and we set macos on a specific config if with clang version X.Y ?

thestr4ng3r commented:
> I throw it just in a simpler way. how about we just have an option? we try with auto, otherwise we provide other optional values and we set macos on a specific config if with clang version X.Y ?

Sorry I don't understand at all what you mean.

wargio commented:
> Sorry I don't understand at all what you mean.

Just an option like `meson -Dblake_cpufeat=avx2` and when is not set (i.e. on `auto`) we try to detect like you did in the code.

thestr4ng3r commented:
Should be possible. But is it really worth it?
If the build with auto works, I don't think there is any good reason to explicitly specify only a strict subset of possible simd implementations (that would be compiled by auto).
If the build with auto fails, specifying `meson -Dblake_cpufeat=` would be an easy workaround, but at the same time it means there is a bug and we should just fix our checking code.
